### PR TITLE
Implementing Unsigned Types

### PR DESCRIPTION
**Pull request by pdet**
Task #1023 

Since the unsigned types are built-in I think the most relevant changes are in the cast operators. Let me know if I forgot any interface or if I should add any other relevant tests.

### CODE REVIEW

**Comment by hannesmuehleisen** [comment]
Very nice, out of curiosity,  how much does this increases the binary size?

**Comment by pdet** [comment]
14mb  with unsigned types, 13mb on master

**Comment by hannesmuehleisen** [comment]
Did you double-check the parquet statistics for the uints are correctly converted?

**Comment by pdet** [comment]
Yes!

The only issue I found with the parquet files was the uint32 being converted to int64.
But that apparently is the expected behavior:
https://issues.apache.org/jira/browse/ARROW-9215?jql=text ~ "uint32"

File: tools/jdbc/src/main/java/org/duckdb/DuckDBResultSet.java [hunk 1]
```diff
@@ -174,6 +183,13 @@ public String getString(int columnIndex) throws SQLException {
 		if ("VARCHAR".equals(meta.column_types[columnIndex - 1])) {
 			return (String) current_chunk[columnIndex - 1].varlen_data[chunk_idx - 1];
 		}
+// 		if ("UBIGINT".equals(meta.column_types[columnIndex - 1])) {
```
Comment (Mytherin) [inline tools/jdbc/src/main/java/org/duckdb/DuckDBResultSet.java:186]: This seems like a leftover

File: test/sql/types/unsigned/test_unsigned_auto_cast.test [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# name: test/sql/types/unsigned/test_unsigned_auto_cast.test
```
Comment (Mytherin) [inline test/sql/types/unsigned/test_unsigned_auto_cast.test:1]: I would like to see more exhaustive tests of casts to VARCHAR and to DECIMAL. Particularly testing boundaries and overflows, e.g.

```sql
SELECT '255'::UTINYINT; -- ok
SELECT '256'::UTINYINT; -- error
SELECT (100::UTINYINT)::DECIMAL(3,0); -- ok
SELECT (100::UTINYINT)::DECIMAL(2,0); --error
```

As well as some tests that verify that decimal arithmetic with auto casts work as expected
```sql
SELECT 100::UTINYINT * 100::DECIMAL(3,0); -- 10000
```

File: test/sql/types/unsigned/test_unsigned_airthmetic.test [hunk 3]
```diff
@@ -0,0 +1,98 @@
+# name: test/sql/types/unsigned/test_unsigned_airthmetic.test
```
Comment (Mytherin) [inline test/sql/types/unsigned/test_unsigned_airthmetic.test:1]: Typo in name: arithmetic

File: test/sql/types/unsigned/test_unsigned_airthmetic.test [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+# description: Test various operations with unsigned
+# group: [unsigned]
+
+statement ok
+PRAGMA enable_verification
+
+#statement ok
+#CREATE TABLE unsigned(a UTINYINT,b USMALLINT, c UINTEGER, d UBIGINT);
```
Comment (Mytherin) [inline test/sql/types/unsigned/test_unsigned_airthmetic.test:9]: Are these tests supposed to be commented out?

File: test/sql/types/unsigned/test_unsigned_auto_cast.test [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+# description: Test unsigned auto cast
```
Comment (Mytherin) [inline test/sql/types/unsigned/test_unsigned_auto_cast.test:2]: I would also like to see some tests involving storage and checkpointing of unsigned types, see e.g. `test/sql/storage/types/test_hugeint_storage.test`. Also including updates to unsigned columns stored in tables.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great, but would like to see some more tests:

**Comment by pdet** [comment]
@Mytherin you got me on the storage. hehe

**Comment by Mytherin** [comment]
Thanks, looks great now.

**Pull request merged**